### PR TITLE
Drop 0.4 support

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: src/BinDeps.jl [hunk 1]
```diff
@@ -3,15 +3,13 @@ __precompile__()
...
 export @make_run, @build_steps, find_library, download_cmd, unpack_cmd,
-        Choice, Choices, CCompile, FileDownloader, FileRule,
-        ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
-        autotools_install, CreateDirectory, MakeTargets, SystemLibInstall
+    Choice, Choices, CCompile, FileDownloader, FileRule,
+    ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
+    autotools_install, CreateDirectory, MakeTargets, SystemLibInstall
 
-const dlext = Libdl.dlext
-const shlib_ext = dlext # compatibility with older packages (e.g. ZMQ)
```
Comment (musm) [inline src/BinDeps.jl]: we should really update all the packages that still depend on this 3 year old definition

Comment (musm) [inline src/BinDeps.jl]: Cairo and HttpParser, still use it

Comment (tkelman) [inline src/BinDeps.jl]: deprecate_binding ?

**Comment by musm** [comment]
done

**Comment by musm** [comment]
I also submitted Prs to update Cairo.jl ,ZMQ.jl and HttpParser

File: src/BinDeps.jl [hunk 2]
```diff
@@ -530,4 +526,10 @@ autotools_install(args...) = error("autotools_install has been removed")
 include("dependencies.jl")
 include("debug.jl")
 include("show.jl")
+
+
+# deprecations
+
+@deprecate shlib_ext Libdl.dlext
```
Comment (tkelman) [inline src/BinDeps.jl]: should use the non-exporting version where possible

Comment (musm) [inline src/BinDeps.jl]: This was wrong I forgot to push the fix, should be g2g now.

**Comment by musm** [comment]
ci failures are unrelated, we are seeing https://github.com/JuliaPackaging/WinRPM.jl/issues/104

File: appveyor.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 environment:
   matrix:
-  - JULIAVERSION: "julialang/bin/winnt/x86/0.4/julia-0.4-latest-win32.exe"
-  - JULIAVERSION: "julialang/bin/winnt/x64/0.4/julia-0.4-latest-win64.exe"
   - JULIAVERSION: "julialang/bin/winnt/x86/0.5/julia-0.5-latest-win32.exe"
   - JULIAVERSION: "julialang/bin/winnt/x64/0.5/julia-0.5-latest-win64.exe"
+  - JULIAVERSION: "julialang/bin/winnt/x86/0.5/julia-0.6-latest-win32.exe"
+  - JULIAVERSION: "julialang/bin/winnt/x64/0.5/julia-0.6-latest-win64.exe"
```
Comment (tkelman) [inline appveyor.yml]: 0.6

**Comment by ihnorton** [comment]
Needs a tag:
https://discourse.julialang.org/t/nettle-fail-to-build/4333/18?u=ihnorton

**Comment by tkelman** [comment]
this was in 0.5.0?

**Comment by ihnorton** [comment]
Oops, I looked at version in the wrong place. Sorry for the noise.

**Pull request merged**